optaplanner-examples/src/main/java/org/optaplanner/examples/examination/solver/score/ExaminationConstraintProvider.java

```diff
@@ -170,30 +185,31 @@ protected Constraint periodSpread(ConstraintFactory constraintFactory) {
                 .join(Exam.class,
                         equal((config, topicConflict) -> topicConflict.getLeftTopic(), Exam::getTopic),
                         filtering((config, topicConflict, leftExam) -> leftExam.getPeriod() != null))
-                .ifExists(Exam.class,
+                .join(Exam.class,
                         equal((config, topicConflict, leftExam) -> topicConflict.getRightTopic(), Exam::getTopic),
                         filtering((config, topicConflict, leftExam, rightExam) -> rightExam.getPeriod() != null),
                         filtering((config, topicConflict, leftExam,
                                 rightExam) -> getPeriodIndexDifferenceBetweenExams(leftExam,
                                         rightExam) < (config.getPeriodSpreadLength() + 1)))
                 .penalizeConfigurable("periodSpread",
-                        (config, topicConflict, leftExam) -> topicConflict.getStudentSize());
+                        (config, topicConflict, leftExam, rightExam) -> topicConflict.getStudentSize());
     }
 
     protected Constraint mixedDurations(ConstraintFactory constraintFactory) {
-        // 4 mixed durations of 100, 150, 200 and 200 should only result in 2 penalties (for 100&150 and 100&200).
```

Comment:
I'd keep this comment.
Otherwise, it's the only reason why this rule is so complex, because of the ITC 2007 spec.
Normally, this would result in 5 penalties, probably mutliplied by the number of students.